src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -4177,7 +4243,12 @@ public FileVisitResult preVisitDirectory(final Path dir, BasicFileAttributes att
                                 return FileVisitResult.CONTINUE;
                             }
                         }
-        
+                        for (File omitFile : omitWatchingFiles) {
+                            if (dir.startsWith(omitFile.getCanonicalPath())) {
```

Comment:
I realized a number of places in DevUtil do something like this, but take for example:
File A has path "/abc/somepath"
File B has path "/abc/somepathwithsubstring/something"
Then `B.startsWith(A.getCanonicalPath())` would return true, even though B is not a child of A.  

The solution could be to check for A's path along with File.separator, or check B.getParent() recursively, but that could be a separate issue since other places in DevUtil have the same issue.  And this is very much a corner case.

Reply:
I've added the check for A's path along with File.separator for this code. As for the rest of DevUtil, we can open a follow up issue if you think it is would be an improvement

Reply:
I'm not sure if File.separator would cause any issues between OS'es.  One scenario that comes to mind is running devc on Windows (but with container's base image using Linux) -- the loose app file is created locally but mounted for the container to use.  I guess this would be fine since dev mode is running locally, but something to test out.